tests/Application/Commands/RedistibuteMatchFundsTest.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+declare(strict_types=1);
+
+namespace MatchBot\Tests\Application\Commands;
+
+use MatchBot\Application\Commands\RedistributeMatchFunds;
+use MatchBot\Application\HttpModels\DonationCreate;
+use MatchBot\Domain\Campaign;
+use MatchBot\Domain\CampaignFunding;
+use MatchBot\Domain\CampaignFundingRepository;
+use MatchBot\Domain\Donation;
+use MatchBot\Domain\DonationRepository;
+use MatchBot\Domain\Pledge;
+use MatchBot\Tests\TestCase;
+use Prophecy\Argument;
+use Prophecy\Prophecy\ObjectProphecy;
+use Psr\Log\NullLogger;
+use Symfony\Component\Console\Tester\CommandTester;
+use Symfony\Component\Lock\LockFactory;
+
+class RedistibuteMatchFundsTest extends TestCase
+{
+    public function testNoEligibleDonations(): void
+    {
+        $donationRepoProphecy = $this->prophesize(DonationRepository::class);
+        $donationRepoProphecy->findWithMatchingWhichCouldBeReplacedWithHigherPriorityAllocation(
+            Argument::type(\DateTimeImmutable::class)
+        )
+            ->willReturn([])
+            ->shouldBeCalledOnce();
```

Comment:
as mentioned on the SUT, it would be great if we can be more specific with the argument than just the type.

Also as this is a query not a command is there any need for `shouldBeCalledOnce()`?